runbotics/runbotics-orchestrator-ui/src/src-app/utils/variableScopes.ts

```diff
@@ -0,0 +1,41 @@
+import { ActionVariableObject, Scope } from '#src-app/hooks/useProcessVariables.types';
+
+export const addScope = (scopeTree: Scope, localScopeId: string, newScopeId: string): void => {
+    if (scopeTree.id === localScopeId) { 
+        scopeTree.children ? scopeTree.children.push({ id: newScopeId }) : scopeTree.children = [{ id: newScopeId }];
+        return;
+    }
+
+    if (!scopeTree.children) return;
+
+    scopeTree.children.forEach(
+        child => addScope(child, localScopeId, newScopeId)
+    );
+};
+
+export const getParentScope = (scopeTree: Scope, scopeId: string): string[] | number => {
```

Comment:
leave a comment that explains return values

Reply:
done